html/semantics/forms/constraints/support/validator.js

```diff
@@ -398,12 +389,27 @@ var validator = {
   iterate_over: function(ctl, data) {
     // Iterate over normal, disabled, readonly, and both.
     var ctlDisabled = ctl.cloneNode(true);
+    this.set_conditions(ctlDisabled, data.conditions);
     ctlDisabled.disabled = true;
+    if (data.dirty)
+      this.set_dirty(ctlDisabled);
```

Comment:
this.set_dirty() should be called before the line of |ctlDisabled.disabled = true;| because set_dirty() relies on focus().

Reply:
Done. I also moved `set_dirty` to before `ctl.readOnly = true;` as `set_dirty` modifies `ctl.value`.